kernelci/data/kernelci_api.py

```diff
@@ -94,6 +103,45 @@ def get_root_node(self, node_id):
         resp = self._get('/'.join(['get_root_node', node_id]))
         return json.loads(resp.text)
 
+    def pubsub_event_filter(self, sub_id, event):
+        """Filter Pub/Sub events
+
+        Filter received Pub/Sub event using provided filter dictionary.
+        Return True if the event matches with the filter, otherwise False.
```

Comment:
The wording might need to be tweaked a bit, because "matching the filter" could be interpreted as "the filter parameters are blocking the event". Say, if the filter parameters match the event data, then the result should be `False` even though it's "matching". Did I get this right?

If that's the case, i.e. `True` means the event should be kept and `False` means the event should be discarded, then the wording should reflect that more clearly.

Reply:
> The wording might need to be tweaked a bit, because "matching the filter" could be interpreted as "the filter parameters are blocking the event". Say, if the filter parameters match the event data, then the result should be `False` even though it's "matching". Did I get this right?
> 
> If that's the case, i.e. `True` means the event should be kept and `False` means the event should be discarded, then the wording should reflect that more clearly.

Actually, the method returns True if no filter dict is provided or event data matches with the filter parameters. Returns False otherwise.

Reply:
I guess "no filter" or an empty dictionary is just a special case where the event is never filtered. So it's consistent with the idea that the method only returns `False` if there's a filter with at least one key that is present in the event data but with a value that doesn't match what is in the event data.